gateway/session_manager.go

```diff
@@ -188,6 +190,12 @@ func (l *SessionLimiter) ForwardMessage(r *http.Request, currentSession *user.Se
 		return sessionFailRateLimit
 	}
 
+	// If MaxQueryDepth is -1 or 0, it means unlimited and no need for depth limiting.
+	if l.DepthLimitEnabled(api.GraphQL.Enabled, accessDef) {
```

Comment:
This code has been moved in a past PR. It now lives in `gateway/mw_graphql_complexity.go`